cmake/CMakeLists.txt

```diff
@@ -50,6 +50,15 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 # NOTE: POSITION INDEPENDENT CODE hurts performance, and it only make sense on POSIX systems
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
+# Suggested by https://gitlab.kitware.com/cmake/cmake/-/issues/20132
+# MacCatalyst does not support well in CMake
+# The error that can emerge without this flag looks like:
+# "clang : error : overriding '-mmacosx-version-min=11.0' option with '-target x86_64-apple-ios12.0-macabi' [-Werror,-Woverriding-t-option]"
+if (PLATFORM_NAME STREQUAL "macabi")
+  add_compile_options(-Wno-overriding-t-option)
+  add_link_options(-Wno-overriding-t-option)
+endif()
+
```

Comment:
adjust_global_compile_flags.cmake may be a better location

'MacCatalyst does not support well in CMake' -> 'MacCatalyst is not well supported in CMake'?

Reply:
moved to adjust_global_compile_flags.cmake instead.